broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/perf/JMHStatePerformanceTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.perf;
+
+import java.text.DecimalFormat;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.openjdk.jmh.results.RunResult;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@EnabledIfEnvironmentVariable(
+    named = "STATE_CONTROLLER_PERFORMANCE_TESTS_ENABLED",
+    matches = "true")
+public class JMHStatePerformanceTest {
+
+  private static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat("0.000");
+  private static final double REFERENCE_SCORE = 0.5;
+  private static final double MAX_DEVIATION = 0.15; // 15% of deviation is allowed
+
+  @Test
+  public void runJmhBenchmark() throws RunnerException {
+    // given
+    final var opt =
+        new OptionsBuilder()
+            .include(LargeStateControllerPerformanceTest.class.getSimpleName())
+            .build();
+
+    // when
+    final var runResults = new Runner(opt).run();
+
+    // then
+    for (final RunResult runResult : runResults) {
+      // we are fine with around 10 percent deviation but not more
```

Comment:
```suggestion
      // we are fine with some deviation but not more than the max
```

Needs to be fixed on the other class as well :)